File: API/Application/DTO/DashboardDto.cs
using API.Core.Models;
using API.Core.Models.Purchases;
using System.Collections;

namespace API.Application.DTO
{
	public class DashboardDto
	{
		public string Currency { get; set; }

		public DashboardAmountDto SalesToday { get; set; }
		public DashboardAmountDto TotalSales { get; set; }
		public DashboardAmountDto AverageSale { get; set; }
		public DashboardAmountDto TotalCustomers { get; set; }

		public Hashtable SalesReport12Month { get; set; }
		public Hashtable SalesReport6Month { get; set; }
		public Hashtable SalesReport30Days { get; set; }
		public Hashtable SalesReport7Days { get; set; }

		public Hashtable TrafficSources12Month { get; set; }
		public Hashtable TrafficSources6Month { get; set; }
		public Hashtable TrafficSources30Days { get; set; }
		public Hashtable TrafficSources7Days { get; set; }

		public Customer[] RecentCustomers { get; set; }

		public Transaction[] RecentTransactions { get; set; }
	}

	public class DashboardAmountDto
	{
		public decimal Amount { get; set; }
		public bool PositiveChange { get; set; }
		public decimal Change { get; set; }
	}
}

File: front-end/src/app/shared/models/dto/report/dashboard-dto.ts
import { TransactionDto } from '../payment/transaction-dto';

export interface Customer {
  photoUrl: string;
  fullName: string;
  email: string;
  currency: string;
  amount: number;
  location: string;
}

export interface DashboardAmountDto {
  amount: number;
  positiveChange: boolean;
  change: number;
}

export interface DashboardDto {
  currency: string;
  salesToday: DashboardAmountDto;
  totalSales: DashboardAmountDto;
  averageSale: DashboardAmountDto;
  totalCustomers: DashboardAmountDto;
  salesReport12Month: { [key: string]: any };
  salesReport6Month: { [key: string]: any };
  salesReport30Days: { [key: string]: any };
  salesReport7Days: { [key: string]: any };
  trafficSources12Month: { [key: string]: any };
  trafficSources6Month: { [key: string]: any };
  trafficSources30Days: { [key: string]: any };
  trafficSources7Days: { [key: string]: any };
  recentCustomers: Customer[];
  recentTransactions: TransactionDto[];
}
